### PR TITLE
Fixed:GCW 2153- display inventory number added

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
               This is PR for GCW-2153- adding inventory number on browse to redirect user to stock app on inventory number Item for Order Fulfilment role user.
https://jira.crossroads.org.hk/browse/GCW-2153

- I have added the fix and also added test case for the same.
- Will be creating separate ticket for the mobile deep linking i.e app to app navigation.
Let me know if any change is required.

Thanks

### CODE REVIEW

File: app/controllers/item.js [hunk 1]
```diff
@@ -81,6 +83,23 @@ export default Ember.Controller.extend({
       this.set('previewUrl', image.get("previewImageUrl"));
     },
 
+    gotoStockItem(inventoryNumber) {
+      let environment = config.staging ? "staging" : config.environment;
+      let finalUrl;
+
+      switch(environment) {
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: Refactor switch case to use **config.APP.BASE_HOST_URL.**
I don't think we need to compare environments here.

File: app/controllers/item.js [hunk 2]
```diff
@@ -81,6 +83,23 @@ export default Ember.Controller.extend({
...
+        case "staging":
+          finalUrl = config.APP.BASE_HOST_URL+"/items/"+inventoryNumber;
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: Also add space before and after "+"

File: app/locales/en/translations.js [hunk 3]
```diff
@@ -123,7 +123,8 @@ export default {
     "quantity": "Quantity",
     "size": "Size",
     "categories": "Categories",
-    "image_unavailable": "No image available."
+    "image_unavailable": "No image available.",
+    "inventory": "Inventory Number"
```
Comment (Darshan-Chauhan) [inline app/locales/en/translations.js]: Will `inventory_number` look much readable here?

File: app/controllers/item.js [hunk 4]
```diff
@@ -81,6 +83,23 @@ export default Ember.Controller.extend({
       this.set('previewUrl', image.get("previewImageUrl"));
     },
 
+    gotoStockItem(inventoryNumber) {
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: A small change **goToStockItem**

File: app/services/session.js [hunk 5]
```diff
@@ -13,6 +13,10 @@ export default Ember.Service.extend({
     return store.peekAll('user').get('firstObject') || null;
   }).volatile(),
 
+  isOrderFulfilmentUser: Ember.computed(function() {
```
Comment (Darshan-Chauhan) [inline app/services/session.js]: This should be method on User model which gives filtered User based on role

File: app/services/session.js [hunk 6]
```diff
@@ -13,6 +13,10 @@ export default Ember.Service.extend({
...
+    return this.get('store').peekAll('role').getEach('name').includes('Order fulfilment');
```
Comment (Darshan-Chauhan) [inline app/services/session.js]: `includes` doesn't work with IE. Use `indexOf`

File: app/models/user.js [hunk 7]
```diff
@@ -21,5 +21,9 @@ export default Addressable.extend({
 
   fullName: Ember.computed('firstName', 'lastName', function(){
     return (this.get('firstName') + " " + this.get('lastName'));
+  }),
+
+  isOrderFulfilmentUser: Ember.computed('roles', function() {
```
Comment (Darshan-Chauhan) [inline app/models/user.js]: If we want to check whether User has some particular role or not we'll have to create computed property for each role... And I think we can refactor it.

May be creating a static function in User model called "hasRole" which takes user object as first argument and role name(string) as second argument and returns truthy value depending or role exists or not.

This should work like rails scopes.

CC: @namrataukirde

File: app/controllers/item.js [hunk 8]
```diff
@@ -81,6 +83,11 @@ export default Ember.Controller.extend({
       this.set('previewUrl', image.get("previewImageUrl"));
     },
 
+    goToStockItem(inventoryNumber) {
+      let finalUrl = config.APP.BASE_HOST_URL + "/items/" + inventoryNumber;
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: Do we really need this variable here??

Comment (abulsayyad123) [reply app/controllers/item.js]: Passing long string as a parameter looks ugly, so assigned it to a variable.

File: config/environment.js [hunk 9]
```diff
@@ -81,14 +81,15 @@ module.exports = function(environment) {
     // RESTAdapter Settings
     ENV.APP.SOCKETIO_WEBSERVICE_URL = 'http://localhost:1337/goodcity';
     ENV.APP.API_HOST_URL = 'http://localhost:3000';
+    ENV.APP.BASE_HOST_URL = 'http://localhost:4203';
```
Comment (Darshan-Chauhan) [inline config/environment.js]: Can we change this name to `STOCK_APP_HOST_URL` for readability and clarity purposes?

Comment (abulsayyad123) [reply config/environment.js]: Yes, that is more clear.

File: app/models/user.js [hunk 10]
```diff
@@ -39,5 +39,17 @@ export default Addressable.extend({
       }
     }
     return hasValue;
+  },
+
+  //*****************
```
Comment (patrixr) [inline app/models/user.js]: I like those comments :)

**Pull request merged**